tests/utils.py

```diff
@@ -72,13 +72,31 @@ def __init__(self, directory="", **files):
             setattr(self, f"{name}_content", content)
 
 
-class AsgiScopeEmulator(dict):
-    """Simulate a real scope. Individual scope values can be overridden by passing
-    a dictionary to the constructor."""
+class AsgiBaseScopeEmulator(dict):
+    """
+    Simulate a basic ASGI scope with minimal default values.
+    Individual scope values can be overridden by passing a dictionary to the constructor.
+    """
 
     def __init__(self, scope_overrides: dict | None = None):
         scope = {
             "asgi": {"version": "3.0"},
+        }
+
+        if scope_overrides:  # pragma: no cover
+            scope.update(scope_overrides)
+
+        super().__init__(scope)
+
+
+class AsgiScopeEmulator(AsgiBaseScopeEmulator):
+    """
+    Simulate a detailed ASGI scope (like HTTP/WS).
+    Individual scope values can be overridden by passing a dictionary to the constructor.
+    """
```

Comment:
```suggestion
class AsgiHttpScopeEmulator(AsgiScopeEmulator):
    """
    Simulate a HTTP ASGI scope.
    Individual scope values can be overridden by passing a dictionary to the constructor.
    """
```

Additionally, will need to change all existing references to this class to `AsgiHttpScopeEmulator`.